frontend/src/stores/search.ts

```diff
@@ -62,36 +63,28 @@ export interface SearchState {
 /**
  * Builds the search query parameters for the given search type, filters, and search term.
  * `q` parameter is always included as the first query parameter.
+ * If the search type is not supported, only the `q` parameter is included.
  * Only the filters that are relevant for the search type and have a value are included.
  *
- * Some parameters are included in the query depending on the mode:
- * - `INCLUDE_SENSITIVE_QUERY_PARAM` is added to the API search query if the setting is `on`
- * in the featureFlagStore.
+ * `INCLUDE_SENSITIVE_QUERY_PARAM` is never added to the frontend search query. It is added
+ * to the API search query if the setting is `on` in the featureFlagStore.
  */
-function computeQueryParams(
+export function computeQueryParams(
   searchType: SearchType,
   filters: Filters,
   searchTerm: string,
   mode: "frontend" | "API"
 ) {
-  // The filters object is converted to a Record<string, string> object.
-  // e.g., { licenseTypes: [{ code: "commercial", checked: true }] }
-  // => { license_type: "commercial" }
-  const query = { ...filtersToQueryData(filters, searchType) }
-
+  if (!isSearchTypeSupported(searchType)) {
+    return { q: searchTerm.trim() }
+  }
   // Ensure that `q` always comes first in the frontend URL.
-  const search_query: ApiQueryParams = { q: searchTerm.trim() }
-
-  // Parameters that are included in the query "as is".
-  const param_names = Object.keys(query).filter(
-    (key): key is Exclude<keyof ApiQueryParams, "q"> =>
-      !["q", INCLUDE_SENSITIVE_QUERY_PARAM].includes(key)
-  )
-
-  for (const api_param_name of param_names) {
-    if (query[api_param_name]?.length) {
-      search_query[api_param_name] = query[api_param_name]
-    }
+  const search_query: SearchQuery = {
```

Comment:
Can we change this to use camel case, while we're modifying this function anyway?

```suggestion
  const searchQuery: SearchQuery = {
```